### PR TITLE
Makes Disarmed Pizzabombs closeable, refactors their code

### LINKED ISSUE #24616: Attempting to open a pizza bomb and cancelling it makes it bug and explode
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

latest as 3/13/2024

### Issue Description

If you double click a pizza bomb it stays closed but opens itself after a while. it gets armed but it appears open

when you try to close it it explodes

### What did you expect to happen?

for the pizza bomb to work correctly

### What happened instead?

it exploded in my face when I tried to close it

### Why is this bad/What are the consequences?

it kills you when it should work in your favor

### Steps to reproduce the issue.

find a pizza bomb, attempt to arm it and cancel it, then it will arm itself but bugged

### When did the problem start happening?

last round, an admin can confirm it

### Extra information

_No response_

### Relevant log output/runtime error

```DM
You have nothing to drop in your hand!
Rihanna Phan says, "What"
 *beep* *beep*
That's no pizza! That's a bomb!
The pizza bomb says, "Enjoy the pizza!"
The pizza bomb violently explodes!
Rihanna Phan has died at Virology. (JMP)
You feel slightly disoriented. That's normal when
```

### LINKED ISSUE #27004: Clicking cancel when prompted for a pizza bomb timer sets the timer anyways.
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1643

### Issue Description

Clicking cancel on the prompt to set a timer for a pizza bomb just sets the timer.

### What did you expect to happen?

I expected the action to be cancelled, allowing me to set a timer later.

### What happened instead?

A timer was set. I blew myself up.

### Why is this bad/What are the consequences?

Blowing myself up on accident is bad.

### Steps to reproduce the issue.

Open pizza bomb. Click cancel on prompt. Close pizza bomb. Open pizza bomb again. 

### When did the problem start happening?

First encountered today.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Refactors 85% of pizza bomb code. Fixes #27004. Fixes #24616.
Pizzas with no tags will no longer have the box_tag overlay.
Pizza bombs boxes can now be toggled closed when counting down to exploding and after they're disarmed.

## Why It's Good For The Game

Gets rid of a lot of bad code. It also adds parity with how normal pizza boxes work given that you can now close the boxes.

## Testing

Normal pizzas seemed fine, and empty pizza boxes had no tags.
Went through the full pizza bomb detonation process. Then sucessfully disarming one, then failing to disarm one and exploding one. Was also able to deconstruct the disarmed one.

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>
  
image
(Changed to be just disarmed pizza bombs)

## Changelog

:cl:
tweak: Disarmed Pizza bombs can now be closed once opened.
fix: Pressing Cancel on a pizza bomb time setting will properly cancel arming it and close the box.
fix: Pizzas with no tags will no longer have the box_tag overlay.
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 1]
```diff
@@ -447,100 +446,140 @@
...
+	var/mob/opener //Ditto
 
-/obj/item/pizzabox/pizza_bomb/update_desc()
-	if(primed)
-		return
+/obj/item/pizzabox/pizza_bomb/Initialize(mapload)
+	correct_wire = pick(wires)
+	var/obj/item/pizzabox/mimic_box = pick(subtypesof(/obj/item/pizzabox) - typesof(/obj/item/pizzabox/pizza_bomb))
+	box_tag = mimic_box.box_tag
+	. = ..()
```
Comment (lewcc) [inline code/modules/food_and_drinks/food/foods/pizza.dm:470]:
```suggestion
	return ..()
```

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 2]
```diff
@@ -447,100 +446,140 @@
...
-			icon_state = "pizzabox1"
+		if(isnull(new_timer) || !in_range(src, user) || issilicon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
+			open = FALSE
+			update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON)
 			return
+
 		timer = new_timer SECONDS
-		icon_state = "pizzabox1"
+		pizza_bomb_status = PIZZA_BOMB_TIMER_SET
+		armer = user
```
Comment (warriorstar-orion) [inline code/modules/food_and_drinks/food/foods/pizza.dm:524]: if the player who armed the bomb leaves the body of the mob that performed the action do we lose the ckey in the logs?

**Comment by Adrer** [comment]
Also fixes #24616 , please include in PR.

File: code/modules/food_and_drinks/food/foods/pizza.dm (hunk 2 above)
Comment (lewcc) [reply code/modules/food_and_drinks/food/foods/pizza.dm:524]: Might be better to hang onto their mind, maybe

**Review by lewcc — APPROVED** [review]
LGTM pending warrior's comment

File: code/modules/food_and_drinks/food/foods/pizza.dm (hunk 2 above)
Comment (Contrabang) [reply code/modules/food_and_drinks/food/foods/pizza.dm:524]: image

**Review by DGamerL — APPROVED** [review]

**Pull request merged**